src/java/org/apache/cassandra/net/MessagingService.java

```diff
@@ -1078,7 +1078,7 @@ public void sendOneWay(MessageOut message, int id, InetAddressAndPort to)
             logger.trace("{} sending {} to {}@{}", FBUtilities.getBroadcastAddressAndPort(), message.verb, id, to);
 
         if (to.equals(FBUtilities.getBroadcastAddressAndPort()))
-            logger.trace("Message-to-self {} going over MessagingService", message);
+            logger.warn("Message-to-self {} going over MessagingService", message);
```

Comment:
Use NoSpamLogger and maybe for the key incorporate the stack trace so we rate limit every call site separately?

Reply:
My last thought is that as a user I can't do much with this information. It's mostly harmless, but I also can't fix it other than by filing a bug report. 

I think this is the biggest issue with this change. It will cause people worry when nothing is wrong. I think this should be debug not warn because no action is required on part of a user (as opposed to a developer).

Reply:
Right, we can do debug here. I wanted to first throw in this case, but then thought that it's more useful to find all the cases where we still do that and eliminate those, since failing in that case brings more or less nothing.